### DISCUSSION TITLE
49. IP 주소와 MAC 주소를 매핑하는 프로토콜에 대해 간략하게 설명해주세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#51] [General]
- IP와 MAC 주소가 각각 무엇인지, 왜 두 주소를 함께 사용하는지 설명해보세요.

`키워드`: ARP

**Comment by cutehumanS2** [comment]
### IP 주소 와 MAC 주소를 매핑하는 프로토콜에 대해 간략하게 설명해주세요.

IP 주소와 MAC 주소를 매핑하는 프로토콜은 ARP(Address Resolution Protocol)입니다. ARP는 동일 네트워크에 속한 호스트의 MAC 주소를 알아내기 위해 사용하는 프로토콜로, IPv4 환경에서 사용됩니다. 

+) ARP: Address Resolution Protocol

<hr>

### IP와 MAC 주소가 각각 무엇인지, 왜 두 주소를 함께 사용하는지 설명해 주세요.
- MAC 주소는 NIC에 할당된 고유한 물리적 주소로, 같은 네트워크(LAN)에 있는 호스트를 식별합니다. 따라서 다른 네트워크의 호스트와 통신할 수 없습니다.
- 반면, IP 주소는 네트워크 상에서 호스트의 논리적 위치를 나타내며, 라우팅을 통해 다른 네트워크와의 통신을 가능하게 합니다.
- 비유하자면, 택배를 보낼 때 수신지(IP 주소)가 먼저 중요하고, 그다음에 수신인(MAC 주소)이 중요한 것처럼, 네트워크 통신에서도 IP 주소를 먼저 확인하여 경로를 결정하고, 목적지 네트워크에 도달한 후 MAC 주소를 사용해 데이터를 최종 호스트로 전달합니다.

**Comment by cutehumanS2** [comment]
### 추가 질문
**[ARP 동작 과정에 대해 설명해 주세요.]**

1. **ARP 요청**: 송신 호스트가 수신 호스트의 IP 주소는 알고 있지만 MAC 주소를 모르는 경우, 네트워크 내의 모든 호스트에게 브로드캐스트로 ARP 요청을 보냅니다.
2. **ARP 응답**: 수신 호스트는 자신의 MAC 주소를 담은 ARP 응답을 송신 호스트에게 유니캐스트로 보냅니다.
3. **ARP 테이블 갱신**: 송신 호스트는 ARP 테이블에 해당 IP와 MAC 주소를 매핑하여 저장하고, 이후 통신에 사용합니다.
<br>

**[다른 네트워크에 속한 호스트에게 패킷을 보내야 할 경우에는 어떻게 하나요?]**
다른 네트워크에 속한 호스트에게 패킷을 보내야 할 경우, 네트워크 외부로 나가기 위한 장비(라우터)의 MAC 주소를 알아내어 패킷을 전송합니다. 


<br>

[참고한 자료]
<혼자 공부하는 네트워크> 03-1. LAN을 넘어서는 네트워크 계층
